### ISSUE TITLE
[FEATURE] automate `branch` and `commit hash` CLI options

### ISSUE THREAD

**Issue by mdjastrzebski** [callstack/reassure#207]
**Is your feature request related to a problem? Please describe.**

Automatically detect branch name and commit hash when `reassure` detects that it is run inside a Git repo. 

**Describe the solution you'd like**

Basically make current `--branch $(git branch --show-current) --commitHash $(git rev-parse HEAD)` options automatic when user is using Git.
Kept the existing CLI options to allow user to override the default values and/or support other source controls systems.
Invoke `git` binary commands using node `child_process`(?) API to avoid external deps.

**Describe alternatives you've considered**
* Instead of invoking external binary use some node package like `simple-git` to communicate with GIT repo. 

**Additional context**

**Comment by ShaswatPrabhat** [comment]
Picking this up please.

**Comment by mdjastrzebski** [comment]
@ShaswatPrabhat Thanks for working on this. I'm not sure I understand your question though.

The data flow between the packages should be the same as is now for branch and commit-hash CLI options. The difference should be that when Reassure 'measure' command does not receive either branch or commit-hash option then it tries to get the missing value(s) from Git if possible.

**Comment by mdjastrzebski** [comment]
A good place for checking these values is the very beginning of "run" method in measure.ts in CLI package

**Comment by ShaswatPrabhat** [comment]
@mdjastrzebski  there were a couple of things I wanted to check:

* Which way of spawning a child process would be preferable ? Right now I am looking at `child_process.fork`, but any notes regarding the same?
* With the above approach the function becomes an IIFE like so:
```(async function () {
  const simpleGit = require('simple-git');
  const git = simpleGit({ baseDir: process.cwd() });
  const branch = await git.revparse(['--abbrev-ref', 'HEAD']);
  const commitHash = await git.revparse(['HEAD']);
  process.send({ branch, commitHash });
})();
```

Please let me know if this is what you had thought of

**Comment by mdjastrzebski** [comment]
I've looked more into the `simple-git` package. It seems that it's actually invoking `git` binary by itself, so we can give it a try.

**Comment by ShaswatPrabhat** [comment]
And an alternative [way](https://github.com/callstack/reassure/pull/211/commits/abd95447ca773ccb517fc59dfa16ecef829eafcd)

**Comment by mdjastrzebski** [comment]
Resolved by #211
